### PR TITLE
add support for non-linear task tree by counting on service to split and merge tasks

### PR DESCRIPTION
**Pull request by Rahien**
add support for non-linear task tree by counting on service to split and merge tasks, readme contains info on how to specify. an example of such a splitter service can be found in https://github.com/lblod/app-decide/pull/88

a convenient service to auto-advance tasks (don't wait for heavy ai services) can be

```
  hacky:
    image: semtech/mu-javascript-template:1.9.1
    environment:
      NODE_ENV: "development"
    volumes:
      - ./config/hacky:/app/
```

app.js
```
import { app } from 'mu';
import { updateSudo as update } from '@lblod/mu-auth-sudo';

app.post('/delta', async function (req, res) {
  res.send('ok');
  completeOpenAnnotationJobs().catch((e) => console.log(e));
});

async function completeOpenAnnotationJobs() {
  await update(`
    PREFIX adms: <http://www.w3.org/ns/adms#>
    PREFIX task: <http://redpencil.data.gift/vocabularies/tasks/>
    DELETE {
     graph ?g {
       ?s adms:status ?status .
     }
    } INSERT {
     graph ?g {
       ?s adms:status <http://redpencil.data.gift/id/concept/JobStatus/success> .
     }
    } WHERE {
     graph ?g {
      VALUES ?ops {
       <http://lblod.data.gift/id/jobs/concept/TaskOperation/codelist-matching/annotate>
       <http://lblod.data.gift/id/jobs/concept/TaskOperation/codelist-matching/train>
      }
      ?s task:operation ?ops ;
         adms:status ?status . 
      FILTER(?status != <http://redpencil.data.gift/id/concept/JobStatus/success>)
     }
    }
    `);
}
```

package.json
```
{
  "name": "hacky",
  "version": "1.0.0",
  "description": "ms for quick hacks.",
  "author": "redpencil.io",
  "license": "MIT",
  "dependencies": {
    "@lblod/mu-auth-sudo": "^1.0.0-beta.4"
  },
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "volta": {
    "node": "20.19.0"
  }
}
```
with delta config

```
{
    match: {
      subject: {},
    },
    callback: {
      url: 'http://hacky/delta',
      method: 'POST',
    },
    options: {
      resourceFormat: 'v0.0.1',
      gracePeriod: 250,
      ignoreFromSelf: true,
      optOutMuScopeIds: [
        'http://redpencil.data.gift/id/concept/muScope/deltas/initialSync',
      ],
    },
  },
```

### CODE REVIEW

**Comment by mirdono** [comment]
I tested this added functionality during the development of our [splitter service](https://github.com/lblod/annotation-job-splitter-service), and it worked as expected. I also did not any breakage with respect to existing functionality, but did not explicitly stress test it.

The code changes look good to me. I don't immediately see potential problems or overlooked corner cases. But I feel I am not yet familiar enough with the service's code base to give a final approval.

File: app.js [hunk 1]
```diff
@@ -2,56 +2,71 @@ import { app, errorHandler } from "mu";
...
-    console.error(`Delta processing failed:`, e.message);
-    return next(e);
-  }
-});
+      return res.status(200).send().end();
+    } catch (e) {
+      console.error(`Delta processing failed:`, e.message);
+      return next(e);
+    }
+  },
```
Comment (nvdk) [inline app.js]: this comma is technically correct, but confusing imo

**Review by nvdk — APPROVED** [review]
